lib/services/files-api-service.js

```diff
@@ -0,0 +1,107 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var fs = require('fs');
+var path = require('path');
+var express = require('express');
+var app = express();
+var walking = require('walk');
+
+module.exports = filesApiServiceFactory;
+
+di.annotate(filesApiServiceFactory, new di.Provide('Http.Services.Api.Files'));
+di.annotate(filesApiServiceFactory,
+    new di.Inject(
+      'Promise',
+      'Services.Waterline'
+    )
+);
+
+
+function filesApiServiceFactory(
+  Promise,
+  waterline
+) {
+  var walker;
+  var walk = Promise.promisifyAll(walking);
+  var onhttpPath = __dirname.split('/');
+  onhttpPath.splice(0,2);
+  onhttpPath = onhttpPath.join("/");
+  Promise.promisifyAll(fs);
+
+  function FilesApiService() {
+  }
+
+  /**
+   * Get list of static files
+   * @returns {array}
+   */
+
+  FilesApiService.prototype.getAllStaticFiles = function() {
+
+    app.use(express.static(onhttpPath));
+    var staticPath = "static/http";
+    var skuPath = "skupack.d";
+    var directories = [staticPath,
+                       skuPath,
+                       "/opt/monorail/static/http"];
+
+    var skuInfo = {};
+    fs.readdirAsync(skuPath).then(function(list) {
```

Comment:
So this whole async call chain isn't related to anything, so essentially it either has no bearing on the rest of the function's code, or is extremely likely to generate race conditions (given that you modify the function-global variable `skuInfo`).